src/Microsoft.ML.Api/DataViewConstructionUtils.cs

```diff
@@ -210,15 +205,9 @@ private Delegate CreateGetter(int index)
                         }
                         else if (outputType == typeof(bool))
                         {
-                            // Bool -> DvBool
-                            Ch.Assert(colType.IsBool);
-                            return CreateConvertingGetterDelegate<bool, DvBool>(index, x => x);
-                        }
-                        else if (outputType == typeof(bool?))
-                        {
-                            // Bool? -> DvBool
+                            // Bool -> Bool.
                             Ch.Assert(colType.IsBool);
-                            return CreateConvertingGetterDelegate<bool?, DvBool>(index, x => x ?? DvBool.NA);
+                            return CreateConvertingGetterDelegate<bool, bool>(index, x => x);
```

Comment:
> return CreateConvertingGetterDelegate<bool, bool>(index, x => x); [](start = 27, length = 66)

not needed, fall back to T-> T default

Reply:
yep.

---
In reply to: [211109149](https://github.com/dotnet/machinelearning/pull/692#discussion_r211109149) [](ancestors = 211109149)